input/drivers/skeleton.py

```diff
@@ -23,16 +23,39 @@ def __init__(self, mapping=None, threaded=True):
             self.mapping = self.default_mapping
         try:
             self.init_hw()
-        except Exception as e:
-            logger.error("init_hw function not found!")
-            logger.exception(e)
+        except AttributeError:
+            logger.error("{}: init_hw function not found!".format(self.__class__))
+        try:
+            self.set_available_keys()
+        except AttributeError:
```

Comment:
ValueError needs to be processed, too

Reply:
... Or not? If a ValueError fails, that means the driver wasn't even tested after being written - so, that should be a high-level failure during testing.